lib/index.ts

```diff
@@ -87,7 +87,8 @@ export default class Backburner {
     };
 
     this._boundRunExpiredTimers = () => {
-      this._runExpiredTimers();
+      this._timerTimeoutId = undefined;
+      this.join(this, this._scheduleExpiredTimers);
```

Comment:
when do we expect this to be wrapped by another run? It should be the systems timer that invokes this right?